### ISSUE TITLE
No DNS with origin containerized install

### ISSUE THREAD

**Issue by jasclarke24** [openshift/openshift-ansible#1593]
I am successfully installing Opemshift origin containerized. but I would like to get. a DNS installed as part of the ansible.  Is there a configuration option for that? Is it supported for origin or containerized?

**Comment by sdodson** [comment]
You mean a dns server that would be used by external entities to resolve hostnames associated with your routes?

**Comment by detiber** [comment]
@jasclarke24 We have the dns role, however it isn't currently called by our general use playbooks and we don't really document how to use it. 

Over the next few months, we should be making it easier to use/consume, but currently we expect that general DNS for the OpenShift hosts and the application domain(s) are handled external to OpenShift.

**Comment by detiber** [comment]
This blog post discusses how dnsmasq can be used for external dns for a cluster: http://developers.redhat.com/blog/2015/11/19/dns-your-openshift-v3-cluster/

**Comment by jasclarke24** [comment]
The problem I am experiencing is in using some of the fabric8 modules
images looking for kubernetes.default.local which they can't resolve.  What
would you recommend as a DNS solution.  set one up externally to openshift
ansible and configure all nodes to it or use what your project has support
for now?  Thanks for your help

**Comment by detiber** [comment]
@jasclarke24 oh, that is a different issue if internal cluster dns isn't resolving.

It should probably be looking for kubernetes.default.svc.cluster.local and not kuberentes.default.local

**Comment by jasclarke24** [comment]
neither is resolvable and the dns config of the containers looks like the
host. How is the internal DNS wires in?  How would I go about checking it?

**Comment by sdodson** [comment]
@jasclarke24 Currently that would only resolve inside a pod that uses ClusterFirst dns, which should be the default. It won't resolve at the host level but I'm working on providing that in https://github.com/openshift/openshift-ansible/pull/1588

**Comment by jasclarke24** [comment]
Is OpenShift-Ansible responsible for setting up the internal CliusterFirst DNS?  If this failed during an Openshift-ansible install any clue as to what was the cause. I do not have any of the related dns containers running (skydns, kube2sky or healthz). I have one etcd running but it sounds like I should have a second one running for dns as outlined here: http://kubernetes.io/docs/admin/dns/
I ran the test unsuccessfully as outlined here: https://github.com/openshift/origin-aggregated-logging/tree/master/deployment.

**Comment by detiber** [comment]
OpenShift uses the same etcd instance for skydns. 

If you do an `oc status` from a master, you should see a line that looks similar to this:
```
svc/kubernetes - 172.30.0.1 ports 443, 53, 53
```

You should be able to reach and query skydns using 172.30.0.1 on udp and tcp port 53 in this case, but if you are overriding the service network, the address may be different.

**Comment by jasclarke24** [comment]
the `oc status` command **did** return the expected results. However we cannot query DNS on 172.30.0.1(53) which is we believe is the correct address in our setup. I just do not believe we have skydns running anywhere in our cluster. Is it a separate container running DNS or a process/service running inside the master container?

**Comment by sdodson** [comment]
SkyDNS is provided by the master service based on config in
/etc/origin/master/master-config.yaml dnsConfig section. If you happen to
have something else bound to the address defined there it could have failed
to start.

**Comment by detiber** [comment]
@jasclarke24 another thing to check for is networking conflicts between the host, pod and service networks.

by default the service network is 172.30.0.0/16 and the pod network is 10.1.0.0/16.

As far as troubleshooting skydns startup, you should be able to see an error in `journalctl -u origin-master -l` if the skydns failed to start as part of the master service startup.

**Comment by tsalopek** [comment]
This issue has been resolved; thank you all for the help. After viewing the logs using the journalctl utility, we found that the DNS was failing to initialize as port 53 was already bound. It appeared as though libvirt had started a dnsmasq instance which in turn bound to port 53. A different, more stripped down centos system image was applied to a different VM which appeared to lack that dnsmasq instance. After going through our openshift installation and configuration, DNS queries resolved as expected.
